Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,694 @@
+            return $order['cart']['order_reference'];
+        }
+
+        $error = __('cart->order_reference does not exist');
+        throw new LocalizedException($error);
+    }
+
+    /**
+     * @param $order
+     * @return string
+     * @throws LocalizedException
+     */
+    public function getDisplayId($order)
+    {
+        if (isset($order['cart']) && isset($order['cart']['display_id'])) {
+            return $order['cart']['display_id'];
+        }
+
+        $error = __('cart->display_id does not exist');
+        throw new LocalizedException($error);
+    }
+
+    /**
+     * Format redirect url taking into account the type of session
+     * the order was created from, front-end vs. back-end
+     *
+     * @param Quote $quote
+     * @return string
+     */
+    public function getReceivedUrl($quote)
+    {
+        $this->sessionHelper->setFormKey($quote);
+
+        $this->logHelper->addInfoLog('[-= getReceivedUrl =-]');
+        $urlInterface = $this->isBackOfficeOrder($quote) ? $this->backendUrl : $this->url;
+        $url = $urlInterface->getUrl('boltpay/order/receivedurl', ['_secure' => true]);
+        $this->logHelper->addInfoLog('---> ' . $url);
+
+        return $url;
+    }
+
+    /**
+     * @param $quote
+     * @return bool
+     */
+    public function isBackOfficeOrder($quote)
+    {
+        return (bool)$quote->getBoltIsBackendOrder();
+    }
+
+    /**
+     * @param $quoteId
+     * @return Quote|null
+     * @throws LocalizedException
+     */
+    public function loadQuoteData($quoteId)
+    {
+        try {
+            /** @var Quote $quote */
+            $quote = $this->cartHelper->getQuoteById($quoteId);
+        } catch (NoSuchEntityException $e) {
+            $this->bugsnag->registerCallback(function ($report) use ($quoteId) {
+                $report->setMetaData([
+                    'ORDER' => [
+                        'pre-auth' => true,
+                        'quoteId' => $quoteId,
+                    ]
+                ]);
+            });
+            $quote = null;
+
+            throw new BoltException(
+                __('There is no quote with ID: %1', $quoteId),
+                null,
+                self::E_BOLT_GENERAL_ERROR
+            );
+        }
+
+        return $quote;
+    }
+
+    /**
+     * @param Quote $quote
+     * @param \stdClass $transaction
+     * @return void
+     * @throws NoSuchEntityException
+     * @throws BoltException
+     */
+    public function validateQuoteData($quote, $transaction)
+    {
+        $this->validateCartItems($quote, $transaction);
+
+        $this->validateTax($quote, $transaction);
+        $this->validateShippingCost($quote, $transaction);
+        $this->validateTotalAmount($quote, $transaction);
+    }
+
+    /**
+     * Get values that are in either array, but not both
+     *
+     * @param array $A
+     * @param array $B
+     * @return array
+     */
+    private function arrayDiff($A, $B) {
```

Comment:
Can we get better naming convention here? Possibly to the use case it is for?

Reply:
@ethanwayda22 Above in phpdoc comment Daniel add proper description for this method. I think it is clear information for anybody who want to use this method.

Reply:
done